interface/procedure_tools/quest/gen_hl7_order.inc.php

```diff
@@ -0,0 +1,590 @@
+<?php
+
+/**
+* Functions to support HL7 order generation.
+*
+* Copyright (C) 2012-2013 Rod Roark <rod@sunsetsystems.com>
+*
+* LICENSE: This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://opensource.org/licenses/gpl-license.php>.
+*
+* @package   OpenEMR
+* @author    Rod Roark <rod@sunsetsystems.com>
+*/
+
+/*
+* A bit of documentation that will need to go into the manual:
+*
+* The lab may want a list of your insurances for mapping into their system.
+* To produce it, go into phpmyadmin and run this query:
+*
+* SELECT i.id, i.name, a.line1, a.line2, a.city, a.state, a.zip, p.area_code,
+* p.prefix, p.number FROM insurance_companies AS i
+* LEFT JOIN addresses AS a ON a.foreign_id = i.id
+* LEFT JOIN phone_numbers AS p ON p.type = 2 AND p.foreign_id = i.id
+* ORDER BY i.name, i.id;
+*
+* Then export as a CSV file and read it into your favorite spreadsheet app.
+*/
+
+require_once("$webserver_root/custom/code_types.inc.php");
+
+use OpenEMR\Common\Logging\EventAuditLogger;
+
+function hl7Text($s)
```

Comment:
Looks like a lot of these functions are just copied across.  I don't think its required you do this for your PR but it'd be great if we could pull these common functions shared across all the gen_hl7_order* files into a shared library.  Maybe open an issue on github and mark it as a newcomer issue for someone to do.

Reply:
@adunsulag Actually what I did was copy the file from labcorp and modified it to work.